src/Containers/Stock.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import styled from 'styled-components';
+import { LineChart, Line } from 'recharts';
+import { ArrowUp } from '@styled-icons/entypo/ArrowUp';
+import { ArrowDown } from '@styled-icons/entypo/ArrowDown';
+import { flex } from '../Utils/Mixins';
+import {
+    Main,
+    Responsive,
+    MainInterface,
+    ResponsiveInterface,
+} from '../Utils/BaseStyles';
+import { MainTheme } from '../Themes';
+
+interface ChartDataItem {
+    value: string | number;
+}
+
+export interface StockProps extends StockBoxProps {
+    chartData: Array<ChartDataItem>;
+    chartColor: string;
+    title: string;
+    figure: number;
+    rate: number;
+    bgColor: string;
+}
+
+const getChartColor = (colorKey: string) =>
+    MainTheme.colors[colorKey] || colorKey;
+
+const getFigure = (figure: number) =>
+    figure.toLocaleString(undefined, { maximumFractionDigits: 2 });
+
+const getRateIcon = (rate: number) =>
```

Comment:
Shouldn't this be inside the component?

Reply:
well, they have no dependencies on props so they can be outside, so I thought why not place them outside, in that way in each render there is no redefinition of the function, but I don't really know where it is better to place them.

Reply:
It is being implemented in the render so it will run each time. Each of the args for the functions are in the list of props. UseCallback would be good here to not run the function on each render if props are the same. I would change getChartColor arg colorKey to chartColor for readability.